pkgs/stdenv/generic/check-meta.nix

```diff
@@ -1,11 +1,11 @@
-# Extend a derivation with checks for brokenness, license, etc.  Throw a
-# descriptive error when the check fails; return `derivationArg` otherwise.
-# Note: no dependencies are checked in this step.
+# Checks derivation meta and attrs for problems (like brokenness,
+# licenses, etc). Calls `config.handleEvalIssue` or `throw` when the
+# check fails; returns `true` otherwise.
 
-{ lib, config, system, meta, derivationArg, mkDerivationArg }:
+{ lib, config, system, meta, derivationArg }:
 
 let
-  attrs = mkDerivationArg; # TODO: probably get rid of passing this one
+  attrs = derivationArg // { inherit meta; };
```

Comment:
Revert this line, or keep it for another patch.  Doing this will actually create a new attribute set based on `derivationArg` only to lookup the `meta` attribute, while doing this on the `mkDerivationArg` will not create any new attribute set.